### PR TITLE
Fixes#8038: Change how status is handled after running workflow

### LINKED ISSUE #8038: Centralise Workflow Status handling
Right now we have copy-paste implementation of the source/sink/stage/bulkink/processor/sink statuses. This should be a base class properly extended instead of duplicating code.

### PR DESCRIPTION
**Pull request by nahuelverdugo**
### Describe your changes:

Fixes #8038

It is changed how we display the status at the end of running a workflow.

When the debug level is DEBUG, we display the OLD style with what is stored on each status:
image

When the debug level is not DEBUG, we display a simple summary with the errors in case there are any: 
image

### Type of change:
- [x] Improvement

### Checklist:
- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`
- [x] I have commented on my code, particularly in hard-to-understand areas. 
- [x] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.
- [x] I have added tests around the new logic.

### CODE REVIEW

**Comment by harshach** [comment]
@nahuelverdugo can we expand the Failure status to include the error and optionally stack trace. It will be easier for us to debug. cc @pmbrull

**Comment by nahuelverdugo** [comment]
> @nahuelverdugo can we expand the Failure status to include the error and optionally stack trace. It will be easier for us to debug. cc @pmbrull

1. We were talking about that today. We can add some kind of verbosity that we want to show, but that would imply sending that parameter somehow from the UI too. I don't mind adding it or opening an issue to add it. What do we want to do?

File: ingestion/src/metadata/utils/workflow_output_handler.py [hunk 1]
```diff
@@ -289,3 +273,133 @@ def print_data_insight_status(workflow) -> None:
...
+            sink,
+            source,
+            stage,
+            source_status,
+            processor_status,
+        )
+    summary = (0, 0, 0, 0)
+    errors = []
+    if source_status and source:
+        summary = tuple(map(sum, zip(summary, get_summary(source_status))))
```
[the remark this answers is not in the document]
Comment (nahuelverdugo) [reply ingestion/src/metadata/utils/workflow_output_handler.py:325]: Let me try to improve it in a way that is easier to understand.

**Comment by harshach** [comment]
@pmbrull @nahuelverdugo so the ask is simple
Expand Failure to not just the name but add other states 
failed {
name: {fqn}
error: {simple error}
stackTrace: {stackTrace}
}

**Comment by nahuelverdugo** [comment]
> @pmbrull @nahuelverdugo so the ask is simple Expand Failure to not just the name but add other states failed { name: {fqn} error: {simple error} stackTrace: {stackTrace} }

We agreed exactly on that.

**Comment by harshach** [comment]
@nahuelverdugo makes sense. Shall we change the call to failure and remove the previous call. I would say
self.status.failed(entityName, errorMessage, stackTrace:optional) ?

**Pull request merged**